catroid/src/org/catrobat/catroid/content/bricks/ShowTextBrick.java

```diff
@@ -0,0 +1,283 @@
+
+		variableSpinner.setAdapter(userVariableAdapterWrapper);
+
+		if (!(checkbox.getVisibility() == View.VISIBLE)) {
+			variableSpinner.setClickable(true);
+			variableSpinner.setEnabled(true);
+		} else {
+			variableSpinner.setClickable(false);
+			variableSpinner.setFocusable(false);
+		}
+
+		setSpinnerSelection(variableSpinner, null);
+
+		variableSpinner.setOnTouchListener(new View.OnTouchListener() {
+
+			@Override
+			public boolean onTouch(View view, MotionEvent event) {
+				if (event.getAction() == MotionEvent.ACTION_UP
+						&& (((Spinner) view).getSelectedItemPosition() == 0
+						&& ((Spinner) view).getAdapter().getCount() == 1)) {
+					NewDataDialog dialog = new NewDataDialog((Spinner) view, NewDataDialog.DialogType.USER_VARIABLE);
+					dialog.addVariableDialogListener(ShowTextBrick.this);
+					dialog.show(((SherlockFragmentActivity) view.getContext()).getSupportFragmentManager(),
+							NewDataDialog.DIALOG_FRAGMENT_TAG);
+					return true;
+				}
+
+				return false;
+			}
+		});
+		variableSpinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+			@Override
+			public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
+				if (position == 0 && ((UserVariableAdapterWrapper) parent.getAdapter()).isTouchInDropDownView()) {
+					NewDataDialog dialog = new NewDataDialog((Spinner) parent, NewDataDialog.DialogType.USER_VARIABLE);
+					dialog.addVariableDialogListener(ShowTextBrick.this);
+					dialog.show(((SherlockFragmentActivity) view.getContext()).getSupportFragmentManager(),
+							NewDataDialog.DIALOG_FRAGMENT_TAG);
+				}
+				((UserVariableAdapterWrapper) parent.getAdapter()).resetIsTouchInDropDownView();
+				userVariable = (UserVariable) parent.getItemAtPosition(position);
+				adapterView = parent;
+			}
+
+			@Override
+			public void onNothingSelected(AdapterView<?> arg0) {
+				userVariable = (UserVariable) adapterView.getItemAtPosition(1);
+
+				userVarName = "No variable set";
+				try {
+					userVarName = userVariable.getName();
+				} catch (NullPointerException e) {
+					Log.d("ShowTextBrick.java", "NullPointerException");
+				}
+			}
+		});
+
+		userVarName = "No variable set";
+		try {
+			userVarName = userVariable.getName();
+		} catch (NullPointerException e) {
+			Log.d("ShowTextBrick.java", "NullPointerException");
+		}
+
+		return view;
+	}
+
+	@Override
+	public View getPrototypeView(Context context) {
+		prototypeView = View.inflate(context, R.layout.brick_show_text, null);
+		TextView textX = (TextView) prototypeView.findViewById(R.id.brick_show_text_prototype_text_view_x);
+		textX.setText(String.valueOf(BrickValues.X_POSITION));
+		TextView textY = (TextView) prototypeView.findViewById(R.id.brick_show_text_prototype_text_view_y);
+		textY.setText(String.valueOf(BrickValues.Y_POSITION));
+		return prototypeView;
+	}
+
+	@Override
+	public View getViewWithAlpha(int alphaValue) {
+
+		if (view != null) {
+
```

Comment:
Remove empty lines

Most of the times you keep a compact and readable coding style. However sometimes you don't. Please be consistent.